### PR TITLE
feat: dedicated stockflow tables

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

- created `tera-stockflow-tables.vue` (code is similar to `tera-petrinet-tables`)
- count for Initial variables appears properly
- an accordion for `flows` has been added
- `auxiliaries` are added to the end of the parameter table as requested by @liunelson 
- `auxiliaries` don't seem to be considered as parameters when the model is converted into MMT so things may not be perfect
- restructured a bit of the file organization

https://github.com/DARPA-ASKEM/terarium/assets/28237258/59c4bf0f-0dcd-4bf7-94c8-ccaf4dfc3461

### CODE REVIEW

File: packages/client/hmi-client/src/components/model/stockflow/tera-stockflow-tables.vue [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+	modelConfigurations?: ModelConfiguration[];
+	readonly?: boolean;
+}>();
+
+const emit = defineEmits(['update-model']);
+
+const mmt = ref<MiraModel>(emptyMiraModel());
+const mmtParams = ref<MiraTemplateParams>({});
+
+const transientModel = ref(cloneDeep(props.model));
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/model/stockflow/tera-stockflow-tables.vue:174]: I think you are cloning this multiple times (e.g. there is one down by the watcher).

File: packages/client/hmi-client/src/components/model/stockflow/tera-stockflow-tables.vue [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+		if (foundParam) {
+			modelAuxiliaries[i] = foundParam;
+		}
+	}
+};
+
+watch(
+	() => props.model,
+	async (model) => {
+		if (!model) return;
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/model/stockflow/tera-stockflow-tables.vue:240]: Your props declaration suggests that `model` is always defined.

File: packages/client/hmi-client/src/components/model/stockflow/tera-stockflow-tables.vue [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+		transientModel.value = cloneDeep(model);
+		const response: any = await getMMT(model);
+		mmt.value = response.mmt;
+		mmtParams.value = response.template_params;
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/model/stockflow/tera-stockflow-tables.vue:244]: If the transient model changes, how do we keep the mmt and mmtParams in sync?

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/components/model/stockflow/tera-stockflow-tables.vue (hunk 3 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/components/model/stockflow/tera-stockflow-tables.vue:243]: Do we want to keep the MMT in sync with the `transientModel` before the edits are saved? I think it makes sense to just sync with the `model` once it's properly updated.

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/components/model/petrinet/tera-petrinet-tables.vue [hunk 4]
```diff
@@ -193,17 +194,21 @@ const updateParam = (params: ModelParameter[]) => {
 	}
 };
 
+async function updateMMT() {
+	const response: any = await getMMT(props.model);
+	mmt.value = response.mmt;
+	mmtParams.value = response.template_params;
+}
```
[the remark this answers is not in the document]
Comment (shawnyama) [reply packages/client/hmi-client/src/components/model/petrinet/tera-petrinet-tables.vue:201]: what's the difference between these?

**Pull request merged**